.github/workflows/main.yml

```diff
@@ -0,0 +1,36 @@
+name: Prebuild Docker image
+
+on:
+  push:
+    branches:
+    - master
+    - cgranade/prebuild-docker # Temporarily enable on wrong branch to help test.
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Azure login
+      uses: Azure/login@v1
+      with:
+        creds: ${{ secrets.ACR_CREDENTIALS }}
+    - name: Docker Login
+      run: az acr login -n msquantum
+    - name: Build and push Docker image
+      run: |
+        $Now = [DateTime]::Now;
+        $ImageTag = "{0}{1:d2}{2:d2}.{3:d2}{4:d2}.{5:d2}{6:d4}" -f $Now.Year, $Now.Month, $Now.Day, $Now.Hour, $Now.Minute, $Now.Second, $Now.Millisecond
```

Comment:
would it be better to use the commit id?

Reply:
Happy to do that as well, it's easily accessible as `${{ github.sha }}`. My only thought was that this gives a strictly increasing ordering. That said, we likely won't need to directly refer to tags other than latest, so let me switch over to SHA sums.